File: backend/src/main/java/com/seonbi/db/entity/Product.java
package com.seonbi.db.entity;

import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

import javax.persistence.*;
import java.util.Objects;

@Entity
@Getter
@Setter
@ToString
@Table(indexes = @Index(name="i_keyword", columnList = "keyword")) // keyword 인덱스로 사용
public class Product extends BaseEntity{

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long productId;
    private Long naverId;
    private String name;
    @Column(length = 1000)
    private String buyUrl;
    private Long price=0l;
    private String imageUrl;
    private String category1;
    private String category2;
    private String category3;
    private String keyword;
    private String brand=null;
    private Long hit=0l;    // buyUrl 눌러서 들어감
    private Long wish=0l;   // 갖고싶소에 넣음
    private Long give=0l;   // 주소싶소에 넣음
    private Long recommend=0l;  // 추천된거


    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Product product = (Product) o;
        return Objects.equals(productId, product.productId) && Objects.equals(naverId, product.naverId) && Objects.equals(name, product.name) && Objects.equals(buyUrl, product.buyUrl) && Objects.equals(price, product.price) && Objects.equals(imageUrl, product.imageUrl) && Objects.equals(category1, product.category1) && Objects.equals(category2, product.category2) && Objects.equals(category3, product.category3)  && Objects.equals(brand, product.brand) && Objects.equals(hit, product.hit) && Objects.equals(wish, product.wish) && Objects.equals(give, product.give) && Objects.equals(recommend, product.recommend);
    }

    @Override
    public int hashCode() {
        return Objects.hash(productId, naverId, name, buyUrl, price, imageUrl, category1, category2, category3, brand, hit, wish, give, recommend);
    }
}

File: frontend/store/interface/product.ts
export class Product {
  id: number
  createdDate: Date
  isDeleted: boolean
  naverId: number
  name: string
  buyUrl: string
  price: number
  imageUrl: string
  category1: string
  category2: string
  category3: string
  brand: string
  hit: number


  constructor(data: any) {
    this.id = data.id || 0
    this.createdDate = data.createdDate || Date.now()
    this.isDeleted = data.isDeleted || false
    this.naverId = data.naverId || 0
    this.name = data.name || ''
    this.buyUrl = data.buyUrl || ''
    this.price = data.price || 0
    this.imageUrl = data.imageUrl || ''
    this.category1 = data.category1 || ''
    this.category2 = data.category2 || ''
    this.category3 = data.category3 || ''
    this.brand = data.brand || ''
    this.hit = data.hit || 0
  }
}

function ProductList(data: Array<any>): Product[] {
  return data.map((now) => {
    return new Product(now)
  })
}

export default ProductList